test/integration_test.py

```diff
@@ -1799,10 +1800,19 @@ class AzulClientIntegrationTest(IntegrationTestCase):
     def test_azul_client_error_handling(self):
         invalid_notification = {}
         notifications = [invalid_notification]
-        self.assertRaises(AzulClientNotificationError,
-                          self.azul_client.index,
-                          first(config.integration_test_catalogs),
-                          notifications)
+        for bundle_notifications_enabled, exception in [
+            (True, AzulClientNotificationError),
+            (False, AzulClientError)
+        ]:
+            with self.subTest(bundle_notifications_enabled=bundle_notifications_enabled):
+                with mock.patch.object(Config,
```

Comment:
PL. Given that this is an integration test, patching the client-side environment to be different from the server-side one is confusing and could lead to failure modes that are difficult to diagnose.

Do we even need to prevent sending the notification in the client? Can't we rely on the 403 or 404 returned by the server?

Reply:
I've removed the patch from IT, and instead assert the response type (400 or 403) based on if the `enable_bundle_notifications` config is enabled on not. I tested this on my personal deployment with the config enabled and then disabled to verify the test succeeds in both cases.